app/controllers/application_controller/policy_support.rb

```diff
@@ -57,8 +57,8 @@ def protect
         add_flash(_("Policy assignments successfully changed"))
         @sb[:action] = nil
       end
-      session[:flash_msgs] = @flash_array
       if @edit[:explorer]
+        @edit = nil if params[:button] == "cancel" || params[:button] == "save"
```

Comment:
@lgalis maybe it can be changed to:
```
if @edit[:explorer] && !(%w(cancel save).include?(params[:button]))
```

Reply:
@ohadlevy, @h-kataria - the condition cannot be changed as suggested, because that would change the action for save/cancel button on the explorer type screens, when we are only looking to clear the @edit hash for these 2 buttons. I will, however use the array of words in my original code change.